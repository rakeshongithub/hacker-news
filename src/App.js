import React, { Suspense, lazy } from "react";
import { BrowserRouter as Router } from "react-router-dom";

import styles from "./App.module.scss";
import LoadingComponent from "./components/UI/Loading";
const AppRoutes = lazy(() => import("./components/Routes/routes"));
const AppHeader = lazy(() => import("./components/Header/AppHeader"));

const App = () => {
  return (
    <Router basename={"/"}>
      <Suspense fallback={<LoadingComponent />}>
        <div className={["container d-flex flex-column", styles.App].join(" ")}>
          <AppHeader />
          <AppRoutes />
        </div>
      </Suspense>
    </Router>
  );
};

export default App;
