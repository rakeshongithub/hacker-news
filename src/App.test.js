import React, { Suspense } from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";

import App, { AppRoutes, AppHeader } from "./App";

describe("App Component", () => {
  it("renders app component without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists()).toBe(true);
  });

  it("renders lazy components", async () => {
    const root = create(
      <Suspense fallback={<div>loading...</div>}>
        <App />
      </Suspense>
    );

    await AppRoutes;
    await AppHeader;
    expect(root).toMatchSnapshot();
  });
});
