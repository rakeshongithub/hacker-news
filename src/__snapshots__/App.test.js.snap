// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders lazy components 1`] = `
<div>
  loading...
</div>
`;
